service/src/main/java/com/ge/predix/acs/privilege/management/dao/GraphGenericRepository.java

```diff
@@ -423,7 +423,9 @@ private void searchAttributesWithScopes(final E entity, final Vertex vertex, fin
 
     private void checkTraversalLimitOrFail(final Set<Attribute> attributes) {
         if (attributes.size() > this.traversalLimit) {
-            throw new QueryException("Graph search failed: traversal limit exceeded.");
+            String exceptionMessage = String.format("Graph search failed: traversal limit of %s exceeded.",
+                    this.traversalLimit);
```

Comment:
We thought about this approach initially, but then this method does not have access to the subject/resource identifier. So, if we just propagate the message upstream, the actual error message sent to the user will not include the subject/resource id. (Unless we add code to catch this exception upstream and add that information). 
For example:
"The number of attributes on this resource/subject /evidence/implant, mulder have exceeded the maximum limit of 256"
vs
"The number of attributes on this resource/subject have exceeded the maximum limit of 256"